From 5fe4fa1c1f426d81496458d2127bfbd7623fe5d5 Mon Sep 17 00:00:00 2001
From: Peter Oh <peter.oh@bowerswilkins.com>
Date: Thu, 12 Apr 2018 02:49:10 -0700
Subject: [PATCH 13/15] mesh: do not allow pri/sec channel switch

We don't want mesh to switch the channel from primary to secondary,
since mesh points are not able to join each other in that case.

Signed-off-by: Peter Oh <peter.oh@bowerswilkins.com>
---
 wpa_supplicant/mesh.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

--- a/wpa_supplicant/mesh.c
+++ b/wpa_supplicant/mesh.c
@@ -333,7 +333,10 @@ static int wpa_supplicant_mesh_init(stru
 			  rate_len * sizeof(int));
 		conf->basic_rates[rate_len] = -1;
 	}
-
+	/* Do not allow primary/secondary channel switch in mesh mode,
+	  * since mesh is not able to establish a physical link for it
+	  */
+	conf->no_pri_sec_switch = 1;
 	wpa_supplicant_conf_ap_ht(wpa_s, ssid, conf);
 
 	if (wpa_drv_init_mesh(wpa_s)) {
